packages/smart-contracts/src/contracts/ERC20TransferrableReceivable.sol

```diff
@@ -0,0 +1,170 @@
+// Contract based on https://docs.openzeppelin.com/contracts/3.x/erc721
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import '@openzeppelin/contracts/utils/Counters.sol';
+import '@openzeppelin/contracts/token/ERC721/extensions/ERC721URIStorage.sol';
+
+contract ERC20TransferrableReceivable is ERC721URIStorage {
+  using Counters for Counters.Counter;
+
+  // Counter for uniquely identifying payments
+  Counters.Counter private _paymentId;
+
+  // Counter for uniquely identifying receivable tokens
+  Counters.Counter private _receivableTokenId;
+
+  struct ReceivableInfo {
+    address tokenAddress;
+    uint256 amount;
+    uint256 balance;
+  }
+  mapping(uint256 => ReceivableInfo) public receivableInfoMapping;
+
+  // Nested mapping for looking up receivable token given a paymentReference
+  // and original payment address
+  mapping(bytes32 => uint256) public receivableTokenIdMapping;
+
+  // Helper mappings to lookup all receivable tokens owned by an address
+  mapping(address => uint256[]) private _receivableIds;
+  mapping(uint256 => uint256) private _receivableIdIndexes;
```

Comment:
Is there a strong reason you implemented your own enumerability instead of using the [ERC721Enumerable OpenZeppelin Extension](https://docs.openzeppelin.com/contracts/4.x/api/token/erc721#IERC721Enumerable)? If so, can you explain?  If not, I think we should use ERC721Enumerable - it provides the standard functions that other contracts and subgraphs expect for quick enumeration.